File: src/ZapCart.WebApi/Models/Product.cs
using System.Collections.Generic;

namespace ZapCart.WebApi.Models
{
    public class Product
    {
        public Product(int id, string name, string description, decimal price, IEnumerable<string> categories)
        {
            Id = id;
            Name = name;
            Description = description;
            Price = price;
            Categories = categories;
        }

        public int Id { get; private set; }

        public string Name { get; private set; }

        public string Description { get; private set; }

        public decimal Price { get; private set; }

        public IEnumerable<string> Categories { get; private set; }
    }
}

File: src/ZapCart/app/scripts/models/product.js
'use strict';

function Product(id, name, description, price) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.price = price;
}

Product.prototype.makeFavorite = function () {
  this.favorite = true;
};

module.exports = Product;